src/v2_vstatefulset_admission_controller.rs

```diff
@@ -0,0 +1,100 @@
+#![allow(unused_imports)]
+
+#[path = "v2/external_shim_layer/mod.rs"]
+pub mod external_shim_layer;
+pub mod kubernetes_api_objects;
+#[path = "v2/kubernetes_cluster/mod.rs"]
+pub mod kubernetes_cluster;
+#[path = "v2/reconciler/mod.rs"]
+pub mod reconciler;
+#[path = "v2/shim_layer/mod.rs"]
+pub mod shim_layer;
+pub mod state_machine;
+pub mod temporal_logic;
+#[path = "v2/controllers/vstatefulset_controller/mod.rs"]
+pub mod vstatefulset_controller;
+pub mod vstd_ext;
+
+use crate::kubernetes_api_objects::exec::dynamic::DynamicObject;
+use crate::vstatefulset_controller::{
+    exec::validator::validate_state, trusted::exec_types::VStatefulSet,
+};
+use deps_hack::anyhow::Result;
+use deps_hack::kube::CustomResourceExt;
+use deps_hack::serde_yaml;
+use deps_hack::tokio;
+// use deps_hack::tracing::{error, info};
+use crate::kubernetes_api_objects::exec::resource::ResourceWrapper;
+use deps_hack::kube::core::{
+    admission::{AdmissionRequest, AdmissionResponse, AdmissionReview},
+    DynamicObject as KubeDynamicObject, ResourceExt,
+};
+use deps_hack::tracing::*;
+use deps_hack::tracing_subscriber;
+use deps_hack::warp::*;
+use shim_layer::controller_runtime::run_controller;
+use std::env;
+// use deps_hack::warp::{reply, Filter, Reply};
+use std::convert::Infallible;
+
+pub async fn validate_handler(
+    body: AdmissionReview<KubeDynamicObject>,
+) -> Result<impl Reply, Infallible> {
+    let req: AdmissionRequest<_> = match body.try_into() {
+        Ok(req) => req,
+        Err(err) => {
+            error!("invalid request: {}", err.to_string());
+            return Ok(reply::json(
+                &AdmissionResponse::invalid(err.to_string()).into_review(),
+            ));
+        }
+    };
+
+    let mut res = AdmissionResponse::from(&req);
+    if let Some(obj) = req.object {
+        let name = obj.name_any();
+        let local_obj = DynamicObject::from_kube(obj.clone());
+
+        // Use unmarshal function to convert DynamicObject to VStatefulSet
+        let vdeploy_result = VStatefulSet::unmarshal(local_obj);
```

Comment:
You might want to rename `vdeploy` to `vstatefulset` for this and other occurrences in this PR